### PR TITLE
Patch 1.2.1: Fix issue where pipeline fails on missing fields from NCBI datasets JSON

### LINKED ISSUE #126: ERROR pipeline failure when fields missing from NCBI genome report [v1.2.0]
### Description of the bug

Key error from `create_table.py`: 
```
Command error:
  Traceback (most recent call last):
    File "/lustre/scratch123/tol/teams/tolit/users/tolpipe/registered_workflows/prod/sanger-tol/genomenote-1.2.0/workflow/bin/create_table.py", line 182, in <module>
      sys.exit(main())
    File "/lustre/scratch123/tol/teams/tolit/users/tolpipe/registered_workflows/prod/sanger-tol/genomenote-1.2.0/workflow/bin/create_table.py", line 171, in main
      ncbi_stats(args.genome, args.sequence, writer)
    File "/lustre/scratch123/tol/teams/tolit/users/tolpipe/registered_workflows/prod/sanger-tol/genomenote-1.2.0/workflow/bin/create_table.py", line 81, in ncbi_stats
      writer.writerow(["Scaffolds", stats["number_of_scaffolds"]])
  KeyError: 'number_of_scaffolds'
```

Failure is on assembly `GCA_963966575.1` which lacks several expected fields in assembly stats: 
```
  "assemblyStats": {
    "contigL50": 32,
    "contigN50": 24519,
    "gcCount": "788905",
    "gcPercent": 38.0,
    "genomeCoverage": "46.0x",
    "numberOfComponentSequences": 87,
    "numberOfContigs": 87,
    "totalSequenceLength": "2065980",
    "totalUngappedLength": "2065980"
  },
  ```

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by tkchafin**
Closes https://github.com/sanger-tol/genomenote/issues/126

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/genomenote/tree/master/.github/CONTRIBUTING.md)
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by muffato — COMMENTED** [review]
Good effort with the linting, @tkchafin, especially adding documentation for all the functions. But unfortunately, `black` disagrees and as you know, it's "opinionated" and "uncompromising" 😬 . We've got `black` installed on the farm in the nf-core Conda environments. That'll make the script comply with the CI.

Then, since we want to make a 1.2.1 release directly with this bugfix, please bump the version and update these files:
- CHANGELOG.md
- nextflow.config
- CITATION.cff
and change the target branch to `main`

**Comment by muffato** [comment]
(the "Run pipeline with test data" CIs now pass. That failure was transient)

File: CITATION.cff [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # This CITATION.cff file was generated with cffinit.
 # Visit https://bit.ly/cffinit to generate yours today!
 
-cff-version: 1.2.0
-title: sanger-tol/genomenote v1.2.0
+cff-version: 1.2.1
```
Comment (muffato) [inline CITATION.cff:4]: Actually I've just realised this version is for the file format itself, which remains 1.2.0. cf https://citation-file-format.github.io/
```suggestion
cff-version: 1.2.0
```

File: bin/create_table.py [hunk 2]
```diff
@@ -6,120 +6,159 @@
 import sys
 import csv
 import re
+import math
 
 
 def parse_args(args=None):
-    Description = "Create a table by parsing json output to extract N50, BUSCO, QV and COMPLETENESS stats."
+    Description = "Create a table by parsing json output to extract N50, " "BUSCO, QV and COMPLETENESS stats."
```
Comment (muffato) [inline bin/create_table.py:13]:
```suggestion
    Description = "Create a table by parsing json output to extract N50, BUSCO, QV and COMPLETENESS stats."
```

**Review by muffato — COMMENTED** [review]
Almost there !

File: CITATION.cff (hunk 1 above)
Comment (tkchafin) [reply CITATION.cff:4]: Oh I see, we were just lucky that it was the same version as the pipeline :)

File: bin/create_table.py (hunk 2 above)
Comment (tkchafin) [reply bin/create_table.py:13]: Ah I think this was introduced by `black` automatic linting from my previous version, which was following PEP8 line length limits. Good to know this is something to watch out for

**Comment by tkchafin** [comment]
Also note that currently we are still using `nan` values as default for all fields. We had discussed using 0 for numeric fields (# chromosomes etc), so I can change it, although I like `nan` since that covers the case that a value is non-zero but erroneously missing in NCBI

**Review by muffato — APPROVED** [review]
Changes look good for me, but can you wait a tiny bit before merging ? We have multiple Zenodo DOIs for our pipelines, and I'm in conversation with them about unifying the DOIs first

**Comment by muffato** [comment]
Merging this as the DOI is actually correct

**Pull request merged**